src/providers/ModalsConditionalsProvider.js

```diff
@@ -0,0 +1,25 @@
+import React, { useState, createContext } from 'react'
+
+export const ModalConditionalsContext = createContext()
+
+// the conditionals for opening a modal
+export const ModalsConditionalsProvider = (props) => {
+
+
```

Comment:
The number of states should correspond directly to the number of modals you have that you can call from different pages. To me it looks like you have more than that.